src/YapepBase/BusinessObject/BoAbstract.php

```diff
@@ -42,17 +42,28 @@
 	 */
 	private $keyPrefix;
 
+	/**
+	 * Indicates if the key storing mechanism is enabled.
+	 *
+	 * @var bool
+	 */
+	private $isKeyStoringEnabled = false;
+
 	/**
 	 * Constructor.
 	 *
 	 * @param string $keyPrefix   The prefix for the cache keys of the bo instance. If not set, one will be generated
 	 *                            based on the configuration and the class-name.
 	 */
 	public function __construct($keyPrefix = null) {
+		$config = Config::getInstance();
+
 		if (empty($keyPrefix)) {
-			$keyPrefix = Config::getInstance()->get('system.project.name') . '.' . get_class($this);
+			$keyPrefix = $config->get('system.project.name') . '.' . get_class($this);
 		}
 		$this->keyPrefix = $keyPrefix;
+
+		$this->isKeyStoringEnabled = $config->get('resource.storage.middleware.isKeyStoringEnabled', false);
```

Comment:
This key is not good. Use system.storage.middleware.isKeyStoringEnabled. `resource` is for named resource configs, like databases that you can have multiple of. `system` is for framework level configs.